lib/synchronised_migration/main.rb

```diff
@@ -60,7 +60,7 @@ def previous_failed?
   end
 
   def mark_failed
-    redis.set fail_key, timestamp, ttl: 3600
+    redis.set fail_key, timestamp, px: 3600
```

Comment:
These times are supposed to be seconds (i.e. fail lock is 1 hour), so using px appears incorrect, see docs:
https://www.rubydoc.info/github/redis/redis-rb/Redis#set-instance_method
Should be using either 'ex' option or convert to milliseconds.

Reply:
indeed I misread the ex keys purpose from the docs